File: app/src/main/java/top/easelink/lcg/ui/main/source/ArticlesDataSource.java
package top.easelink.lcg.ui.main.source;

import androidx.annotation.NonNull;

import java.util.List;

import javax.annotation.Nullable;

import io.reactivex.Observable;
import top.easelink.lcg.ui.main.source.model.Article;
import top.easelink.lcg.ui.main.source.model.ArticleDetail;
import top.easelink.lcg.ui.main.source.model.ForumPage;

/**
 * author : junzhang
 * date   : 2019-07-26 14:59
 * desc   :
 */
public interface ArticlesDataSource {

    Observable<ForumPage> getForumArticles(@NonNull final String requestUrl, boolean processThreadList);

    Observable<ArticleDetail> getArticleDetail(@NonNull final String url);

    Observable<List<Article>> getHomePageArticles(@NonNull final String param, @Nullable final Integer pageNum);

}

File: app/src/main/java/top/easelink/lcg/ui/main/source/ArticlesDataSource.kt
package top.easelink.lcg.ui.main.source

import androidx.annotation.WorkerThread
import top.easelink.lcg.ui.main.source.model.Article
import top.easelink.lcg.ui.main.source.model.ArticleDetail
import top.easelink.lcg.ui.main.source.model.ForumPage
import top.easelink.lcg.ui.main.source.model.PreviewPost

/**
 * author : junzhang
 * date   : 2019-07-26 14:59
 * desc   :
 */
interface ArticlesDataSource {
    @WorkerThread
    fun getForumArticles(query: String, processThreadList: Boolean): ForumPage?

    @WorkerThread
    fun getArticleDetail(query: String, isFirstFetch: Boolean): ArticleDetail?

    @WorkerThread
    fun getPostPreview(query: String): PreviewPost?

    @WorkerThread
    fun getHomePageArticles(param: String, pageNum: Int): List<Article>
}